packages/routers/src/__tests__/StackRouter.test.tsx

```diff
@@ -2497,6 +2499,7 @@ test('handles screen preloading', () => {
     ],
     routeNames: ['baz', 'bar', 'qux'],
     routes: [
+      { key: 'qux-some', name: 'qux' },
       {
         key: 'bar-test',
```

Comment:
The `StackAction.push` should push a new route, even if the IDs match with a previous route in history.